ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_pim_sparse_mode.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdMissingVariableError
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import default, get
+from ansible_collections.arista.avd.roles.eos_designs.python_modules.network_services.utils import UtilsMixin
+
+
+class RouterPimSparseModeMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def router_pim_sparse_mode(self) -> dict | None:
+        """
+        return structured config for router_pim
+
+        Used for to configure RPs on the VRF
+        """
+
+        if not self._network_services_l3:
+            return None
+
+        vrfs = []
+        for tenant in self._filtered_tenants:
+            for vrf in tenant["vrfs"]:
+                rps = []
+                for node_item in (mc_node_settings := default(get(vrf, "l3_multicast.node_settings"), get(tenant, "l3_multicast.node_settings"), [])):
+                    if self._hostname in (mc_nodes := get(node_item, "nodes", required=False, default=[])) or not mc_nodes:
+                        if not mc_nodes and len(mc_node_settings) > 1:
+                            raise AristaAvdMissingVariableError(
+                                f"l3_multicast.node_settings in Tenant '{tenant['name']}' or VRF '{vrf['name']}': only one entry with no 'nodes' or multiple"
+                                " entries with 'nodes' can be defined."
+                            )
+
+                        for rp_item in get(node_item, "rp_addresses"):
+                            for rp_address in get(
+                                rp_item,
+                                "rps",
+                                required=True,
+                                org_key=f"l3_multicast.node_settings.rp_addresses.rps under VRF '{vrf['name']}' in tenant '{tenant['name']}'",
```

Comment:
so what is the reason behind this - is the returned key error not clear enough?

If not, I think it would be better to return something that "looks" like a key rather than a string so maybe something like:

```suggestion
                                org_key=f"tenants[{tenant['name']}].vrfs[{vrf['name']}].l3_multicast.node_settings.rp_addresses.rps",
```

Reply:
Just following behaviour seen elsewhere.  For example a change of Claus' in `vxlan_interface.py`:
```
org_key=f"'mac_vrf_vni_base' for Tenant: {tenant['name']}"
```